packages/plugins/model-train/image-classification-tfjs-model-train/src/index.ts

```diff
@@ -3,25 +3,36 @@ import { ImageDataset, ModelTrainType, UniModel, ModelTrainArgsType, ImageDataLo
 import * as tf from '@tensorflow/tfjs-node-gpu';
 import Jimp from 'jimp';
 
+async function dataIterator(dataLoader: ImageDataLoader, labelMap: {
+  [key: string]: number;
+}): Promise<any> {
+  const count = await dataLoader.len();
+  return (): AsyncIterator<tf.TensorContainer> => {
+    let index = 0;
+    return {
+      async next(): Promise<IteratorResult<tf.TensorContainer>> {
+        if (index >= count) {
+          return { value: null, done: true };
+        }
+        const currentData = await dataLoader.getItem(index);
+        const image = await Jimp.read(currentData.data);
+        const trainImageBuffer = await image.getBufferAsync(Jimp.MIME_JPEG);
+        const imageArray = new Uint8Array(trainImageBuffer);
+        const label: number = currentData.label.categoryId;
+        index++;
+        return { value: tf.tidy(() => ({
+          xs: tf.tidy(() => tf.cast(tf.node.decodeImage(imageArray, 3), 'float32')),
+          ys: tf.tidy(() => tf.oneHot(tf.scalar(label, 'int32'), Object.keys(labelMap).length))
```

Comment:
Could we refactor `labelMap ` to `Map` from 'object'?

Reply:
Could give a new PR to do this refractor, also include other plugins.